### PR TITLE
Migrate to CFT logging modules

### PR DESCRIPTION
**Pull request by umairidris**
I have upstreamed features from DPT into the CFT logging modules so they mostly match our feature set. They also have new features to set things like bucket locks which are useful for hipaa alignment.

### CODE REVIEW

**Review by MartinPetkov — APPROVED** [review]

File: templates/tfengine/components/audit/main.tf [hunk 1]
```diff
@@ -40,84 +40,67 @@ resource "google_{{.parent_type}}_iam_audit_config" "config" {
...
-
-  dataset_id                  = "{{.logs_bigquery_dataset.dataset_id}}"
-  project_id                  = module.project.project_id
-  location                    = "{{.bigquery_location}}"
-  default_table_expiration_ms = 365 * 8.64 * pow(10, 7) # 365 days
-  access = [
-    {
-      role          = "roles/bigquery.dataOwner",
-      special_group = "projectOwners"
-    },
```
Comment (xingao267) [inline templates/tfengine/components/audit/main.tf:64]: do we still grant this permissions separately?

File: templates/tfengine/components/audit/main.tf [hunk 2]
```diff
@@ -40,84 +40,67 @@ resource "google_{{.parent_type}}_iam_audit_config" "config" {
...
+  location                 = "{{.bigquery_location}}"
+  log_sink_writer_identity = "${module.bigquery_export.writer_identity}"
+  expiration_days          = 365
 }
 
-resource "google_project_iam_member" "bigquery_sink_member" {
-  project = module.bigquery_destination.bigquery_dataset.project
-  role    = "roles/bigquery.dataEditor"
-  member  = google_logging_{{.parent_type}}_sink.bigquery_audit_logs_sink.writer_identity
-}
```
Comment (xingao267) [inline templates/tfengine/components/audit/main.tf:76]: do we need this or it's now part of the log-export module?

File: templates/tfengine/components/audit/main.tf [hunk 3]
```diff
@@ -40,84 +40,67 @@ resource "google_{{.parent_type}}_iam_audit_config" "config" {
...
+
+  storage_bucket_name      = "{{.logs_storage_bucket.name}}"
+  project_id               = module.project.project_id
+  location                 = "{{.storage_location}}"
+  log_sink_writer_identity = "${module.storage_export.writer_identity}"
+  storage_class            = "COLDLINE"
+  expiration_days          = 7*365
+  retention_policy = {
+    is_locked             = true
+    retention_period_days = 6 * 365
```
Comment (xingao267) [inline templates/tfengine/components/audit/main.tf:92]: what's the different between the `retention_period_days` here and `expiration_days` above? and why are they set to different values?

File: templates/tfengine/components/audit/main.tf [hunk 4]
```diff
@@ -40,84 +40,67 @@ resource "google_{{.parent_type}}_iam_audit_config" "config" {
...
-  ]
+  source  = "terraform-google-modules/log-export/google//modules/storage"
+  version = "~> 5.0.0"
+
+  storage_bucket_name      = "{{.logs_storage_bucket.name}}"
+  project_id               = module.project.project_id
+  location                 = "{{.storage_location}}"
+  log_sink_writer_identity = "${module.storage_export.writer_identity}"
+  storage_class            = "COLDLINE"
+  expiration_days          = 7*365
```
Comment (xingao267) [inline templates/tfengine/components/audit/main.tf:89]: nit: `7 * 365` (with the spaces)

File: templates/tfengine/components/audit/main.tf (hunk 1 above)
Comment (umairidris) [reply templates/tfengine/components/audit/main.tf:64]: dataViewer is done by the project iam permission below (made it bigquery.user)

Converting BQ permissions away from default is being tracked by https://github.com/terraform-google-modules/terraform-google-log-export/issues/49

File: templates/tfengine/components/audit/main.tf (hunk 2 above)
Comment (umairidris) [reply templates/tfengine/components/audit/main.tf:76]: This one's done by the cft module

File: templates/tfengine/components/audit/main.tf (hunk 4 above)
Comment (umairidris) [reply templates/tfengine/components/audit/main.tf]: Done.

File: templates/tfengine/components/audit/main.tf (hunk 3 above)
Comment (umairidris) [reply templates/tfengine/components/audit/main.tf:95]: Retention period does not actually do anything. It just forces the bucket to permanently have an expiry that is at least this amount. See https://cloud.google.com/storage/docs/bucket-lock. I also added a comment.

p.s. there seems to be an org policy we should probably set as well: https://cloud.google.com/storage/docs/org-policy-constraints#audit-logging. Do you want to look into that?

Comment (xingao267) [reply templates/tfengine/components/audit/main.tf:95]: sure i'll look into that.

**Review by xingao267 — APPROVED** [review]

**Pull request merged**